### ISSUE TITLE
Error missing expected { when updating the provider

### ISSUE THREAD

**Issue by igoratencompass** [mongodb/terraform-provider-mongodbatlas#697]
When upgrading the provider version like this:

```
    mongodbatlas = {
      source  = "mongodb/mongodbatlas"
-     version = "~> 0.9.1"
+     version = "~> 1.0.1"
    }
```

we end up with the following error during the plan phase:

```
│ Error: missing expected {
│ 
│   with module.mongodb-atlas["env"].mongodbatlas_cloud_provider_access.atlas[0],
│   on ../terraform-modules/modules/mongodb-atlas/main.tf line 195, in resource "mongodbatlas_cloud_provider_access" "atlas":
│  195: resource "mongodbatlas_cloud_provider_access" "atlas" {
│ 
```

Please see this link https://stackoverflow.com/questions/68648971/terraform-plan-error-missing-expected where an issue with data schema migration is mentioned as probable cause of the error. Modifying the state file as explained in the link fixes the plan issue.

Thanks

**Comment by themantissa** [comment]
@igoratencompass thank you for the report. We'll take a look at this.

**Comment by themantissa** [comment]
Internal ticket INTMDB-309

**Comment by themantissa** [comment]
@igoratencompass  this is corrected in version 1.4.0, which is now available as a pre-release version in this repo.  We will release it tomorrow.

**Comment by themantissa** [comment]
@igoratencompass this is now out in 1.4.1.
